File: hadiah_api/app/Models/Hadiah.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Hadiah extends Model
{
    use HasFactory;

    protected $fillable = ['toko_id', 'doc_number', 'received', 'received_date', 'failed_reason'];

    public function toko()
    {
        return $this->belongsTo(Toko::class);
    }
}

File: hadiah_app/lib/models/hadiah.dart
class Hadiah {
  final String toko;
  final String hadiah;
  final String tanggal;
  final bool received;
  final String failedReason;

  Hadiah({
    required this.toko,
    required this.hadiah,
    required this.tanggal,
    required this.received,
    required this.failedReason,
  });

  factory Hadiah.fromJson(Map<String, dynamic> json) {
    return Hadiah(
      toko: json['toko'],
      hadiah: json['hadiah'],
      tanggal: json['tanggal'],
      received: json['received'],
      failedReason: json['failedReason'],
    );
  }
}
